packages/block-editor/src/components/navigable-toolbar/index.js

```diff
@@ -28,42 +33,69 @@ class NavigableToolbar extends Component {
 
 	componentDidMount() {
 		if ( this.props.focusOnMount ) {
-			this.focusToolbar();
+			window.requestAnimationFrame( () => {
+				this.focusToolbar();
+			} );
 		}
 
-		// We use DOM event listeners instead of React event listeners
-		// because we want to catch events from the underlying DOM tree
-		// The React Tree can be different from the DOM tree when using
-		// portals. Block Toolbars for instance are rendered in a separate
-		// React Tree.
-		this.toolbar.current.addEventListener( 'keydown', this.switchOnKeyDown );
-	}
+		// Toolbar items added via Portal (Slot bubblesVirtually) aren't added
+		// to the DOM right away
+		window.requestAnimationFrame( () => {
```

Comment:
@youknowriad  and @aduth - do you know if we can avoid it?

Reply:
It's not ideal that they're not made available immediately. I'm unsure if the implementation of Slot/Fill can be changed to support what's need here.

I've not yet looked at the rest of the code proposed here, but one option to consider is if it might be possible to apply a `componentDidMount` to the elements _within_ the fill, to use as an indicator of when they're actually available in the DOM. I expect it could make things more challenging to coordinate (multiple fill elements, coordinating with an ancestor component).